File: backend/ria_ml/SmithCase_Chapter15.m
% SmithCase_Chapter15.m

% clear all previous variables and close any figures
   clear all;
   close all;
   
% create a new client data structure
   client = client_create();
% process the client data structure 
   client = client_process(client);
   
% create a new market data structure
   market = market_create( );
   
% process the client data structure
   market = market_process(market,client); 

% create and process AMDnLockboxes
   AMDnLockboxes = AMDnLockboxes_create( );
   AMDnLockboxes.showProportions = 'y';
   AMDnLockboxes = AMDnLockboxes_process(AMDnLockboxes, market, client);
     
% create and process CMULockboxes
   CMULockboxes = CMULockboxes_create( );
   CMULockboxes.showProportions = 'y';
   CMULockboxes = CMULockboxes_process(CMULockboxes, market, client);
  

% create and process combined lockboxes
   combinedLockboxes = combinedLockboxes_create( );
   combinedLockboxes.componentLockboxes = {AMDnLockboxes CMULockboxes};
   combinedLockboxes.componentWeights = [0.5 0.5];
   combinedLockboxes.showCombinedProportions = 'y';
   combinedLockboxes = combinedLockboxes_process(combinedLockboxes, client); 
   

File: backend/ria_ml/SmithCase_Chapter15.py
from client_create import client_create
from client_process import client_process
from market_create import market_create
from market_process import market_process
from AMDnLockboxes_create import AMDnLockboxes_create
from AMDnLockboxes_process import AMDnLockboxes_process
from CMULockboxes_create import CMULockboxes_create
from CMULockboxes_process import CMULockboxes_process
from combinedLockboxes_create import combinedLockboxes_create
from combinedLockboxes_process import combinedLockboxes_process

client = client_create()
client = client_process(client)

market = market_create()
market = market_process(market, client)

AMDnLockboxes = AMDnLockboxes_create()
AMDnLockboxes['showProportions'] = 'y'
AMDnLockboxes = AMDnLockboxes_process(AMDnLockboxes, market, client)

CMULockboxes = CMULockboxes_create()
CMULockboxes['showProportions'] = 'y'
CMULockboxes = CMULockboxes_process(CMULockboxes, market, client)

combinedLockboxes = combinedLockboxes_create()
combinedLockboxes['componentLockboxes'] = [AMDnLockboxes, CMULockboxes]
combinedLockboxes['componentWeights'] = [0.5, 0.5]
combinedLockboxes['showCombinedProportions'] = 'y'
combinedLockboxes = combinedLockboxes_process(combinedLockboxes, client)
